components/post_view/post_image/post_image.jsx

```diff
@@ -60,8 +60,9 @@ export default class PostImageEmbed extends React.PureComponent {
         };
     }
 
-    UNSAFE_componentWillMount() { // eslint-disable-line camelcase
+    componentDidMount() {
         this.loadImg(this.props.link);
```

Comment:
Moving this from componentWillMount to componentDidMount will delay the request going out until mounting completes. Would it make sense to move this to the constructor instead? Not sure if the delay will be noticeable but a thought I had

Reply:
componentWillMount is actually deprecated and even if you initiate the request in componentWillMount it will not return until the component has mounted.

there is more info here https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#fetching-external-data

Reply:
> There is a common misconception that fetching in componentWillMount lets you avoid the first empty rendering state. In practice this was never true because React has always executed render immediately after componentWillMount. If the data is not available by the time componentWillMount fires, the first render will still show a loading state regardless of where you initiate the fetch. This is why moving the fetch to componentDidMount has no perceptible effect in the vast majority of cases.

Reply:
@enahum JW is suggesting to move it to constructor instead of didMount.

JW suggestion got me thinking. Should we be doing things like this in constructor or on mount? I know either way the performance benefit would be really small. 

It brings up a good point that we might need to resort to componentDidMount at times for heavier tasks so we can mount the component faster for perceived performance other than this i think we can resort to having them in constructor. In this case constructor

Reply:
The constructor should no be performing any sort of requests and having in the constructor won’t prevent the component from rendering anyway and the fact that requests are async you won’t get any performance gains

Reply:
There is a new api for suspending and resuming rendering but I wouldn’t go in that direction just yet

Reply:
I know it won't prevent the first empty rendering, I wasn't saying that at all. Firing off the request in the constructor will have it start early though, instead of waiting for the component to mount before starting the request.